processor/marketplace_processor/holding/holding_creation.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2017 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# -----------------------------------------------------------------------------
+
+from sawtooth_sdk.processor.exceptions import InvalidTransaction
+
+
+def handle_holding_creation(create_holding, header, state):
+    """
+
+    Args:
+        create_holding (CreateHolding): The transaction.
+        header (TransactionHeader): The header of the Transaction.
+        state (MarketplaceState): The wrapper around the context.
+
+    Raises:
+        InvalidTransaction
+            - There is already a Holding with the same identifier.
+            - The account does not exist.
+            - The account holder is the Txn signer.
+            - The asset/holding does not exist.
+            - The quantity is not 0 and the Asset owner doesn't match the
+              transaction signer public key.
+    """
+
+    if state.get_holding(identifier=create_holding.id):
+        raise InvalidTransaction("Failed to create Holding, id {} already "
+                                 "exists.".format(create_holding.id))
+
+    account = state.get_account(public_key=create_holding.account)
+    if not account:
+        raise InvalidTransaction(
+            "Failed to create Holding, account {} does not exist.".format(
+                create_holding.account))
+
+    if account.public_key != header.signer_public_key:
+        raise InvalidTransaction(
+            "Failed to create Holding, account {} does not belong to "
+            "txn signer {}".format(account.public_key,
+                                   header.signer_public_key))
+
+    asset = state.get_asset(name=create_holding.holding)
```

Comment:
See typo comment above.

Reply:
I will update the CreateHolding proto. It is assumed the the txn signer is creating the Holding for their account, right?

Reply:
Yes. They will always be creating holdings for their own account.